### PR TITLE
[CI] Move tests outside of the build machine.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: tools/devops/automation/build-pipeline.yml [hunk 1]
```diff
@@ -287,21 +304,41 @@ stages:
 - ${{ if and(ne(variables['Build.Reason'], 'Schedule'), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), eq(parameters.forceInsertion, true))) }}:
   - template: templates/release/vs-insertion-prep.yml
 
-# Test stages
+# Test stages#
```
Comment (tj-devel709) [inline tools/devops/automation/build-pipeline.yml:307]:
```suggestion
# Test stages
```

File: tools/devops/automation/build-pipeline.yml [hunk 2]
```diff
@@ -287,21 +304,41 @@ stages:
...
+      testsLabels: '' # hardcoded has to be fixed, run all the things, we can split later
+      statusContext: ${{ config['statusContext'] }} 
+      iOSDeviceDemand: ${{ config['iOSDeviceDemand'] }} 
+      makeTarget: ${{ config['makeTarget'] }} 
+      vsdropsPrefix: ${{ variables.vsdropsPrefix }}
+      keyringPass: $(pass--lab--mac--builder--keychain)
+      gitHubToken: ${{ variables['GitHub.Token'] }}
+      xqaCertPass: $(xqa--certificates--password)
+
+# devices are optinal and will only be ran when we set them OR in CI
```
Comment (tj-devel709) [inline tools/devops/automation/build-pipeline.yml:327]:
```suggestion
# devices are optional and will only be ran when we set them OR in CI
```

File: tools/devops/automation/templates/tests/build.yml [hunk 3]
```diff
@@ -43,6 +43,10 @@ parameters:
 - name: xqaCertPass
   type: string
 
+- name: makeTarget
+  type: string
+  default: 'vsts-device-tests' # target to be usde to run the tests
```
Comment (tj-devel709) [inline tools/devops/automation/templates/tests/build.yml:48]:
```suggestion
  default: 'vsts-device-tests' # target to be used to run the tests
```

File: tools/devops/automation/templates/tests/run-tests.yml [hunk 4]
```diff
@@ -18,6 +18,10 @@ parameters:
   type: string
   default: 'ios' # default context, since we started dealing with iOS devices.
 
+- name: makeTarget
+  type: string
+  default: 'vsts-device-tests' # target to be usde to run the tests
```
Comment (tj-devel709) [inline tools/devops/automation/templates/tests/run-tests.yml:23]:
```suggestion
  default: 'vsts-device-tests' # target to be used to run the tests
```

File: tools/devops/automation/templates/tests/stage.yml [hunk 5]
```diff
@@ -58,6 +55,10 @@ parameters:
 - name: xqaCertPass
   type: string
 
+- name: makeTarget
+  type: string
+  default: 'vsts-device-tests' # target to be usde to run the tests
```
Comment (tj-devel709) [inline tools/devops/automation/templates/tests/stage.yml:60]:
```suggestion
  default: 'vsts-device-tests' # target to be used to run the tests
```

File: tools/devops/automation/build-pipeline.yml [hunk 6]
```diff
@@ -70,33 +70,50 @@ parameters:
...
+    {
+      devicePrefix: 'simulator',
+      stageName: 'simulator',
+      displayName: 'Simulator Tests',
+      iOSDevicePool: '', # TODO empty because we want to use the one used by the agent pool selector
+      useXamarinStorage: False,
+      testsLabels: '--label=run-all-tests,skip-device-tests', # TODO same as calling jenkins, we need to use diff one in ci an 
+      statusContext: 'VSTS: simulator tests iOS',
+      makeTarget: 'jenkins',
+      iOSDeviceDemand: [] # we do no need extra capabilities
```
Comment (tj-devel709) [inline tools/devops/automation/build-pipeline.yml:87]:
```suggestion
      iOSDeviceDemand: [] # we do not need extra capabilities
```

**Review by tj-devel709 — COMMENTED** [review]
Looks to go me so far! Found a few small typos!

**Review by tj-devel709 — COMMENTED** [review]
Few fluff comments but the one about the lines that should be comments is probably significant!

**Review by dalexsoto — APPROVED** [review]
🐥📼 ✅

**Review by rolfbjarne — APPROVED** [review]

**Review by chamons — APPROVED** [review]

**Pull request merged**